src/main/java/com/google/api/codegen/util/js/JSCommentReformatter.java

```diff
@@ -38,32 +36,7 @@ public String reformat(String comment) {
   }
 
   public String getLinkedElementName(ProtoElement element) {
-    if (isExternalFile(element.getFile())) {
-      String fullName = element.getFullName();
-      return String.format("[%s]{@link external:\"%s\"}", fullName, fullName);
-    } else {
-      String simpleName = element.getSimpleName();
-      return String.format("[%s]{@link %s}", simpleName, simpleName);
-    }
+    String simpleName = element.getSimpleName();
+    return String.format("[%s]{@link %s}", simpleName, simpleName);
```

Comment:
`return String.format("[%s]{@link %s.%s}", simpleName, packageName, simpleName);`

Reply:
I think using `element.fullName` will get us what we want. This will change the all of the type links.

Reply:
JK I think what you have is correct!

Reply:
I think we need to use `packageName`. For nested messages, `fullName` will contain an extra path element that JSDoc won't be able to resolve. For example

```proto
package "farm/animals"

message Horse {
  message Thing {}
}

message Cow {
}
```

```js
/**
 * @typedef Horse
 * @memberof farm.animals
 */
var Horse = {
  /**
   * @typedef Thing
   * @memberof farm.animals
   */ 
  Thing: {}
};
/**
 * @typedef Cow
 * @memberof farm.animals
 */
var Cow = {};
```

For `Thing`, `element.fullName` would try to link to it using `farm.animals.Horse.Thing`, but JSDoc wants to link to it using `farm.animals.Thing`.